internal/translation/deployment/deployment.go

```diff
@@ -215,3 +232,69 @@ func (ds *ProductionAtlasDeployments) UpdateProcessArgs(ctx context.Context, clu
 
 	return nil
 }
+
+func (ds *ProductionAtlasDeployments) GetCustomZones(ctx context.Context, projectID, clusterName string) (map[string]string, error) {
+	geosharding, _, err := ds.globalClusterAPI.GetManagedNamespace(ctx, projectID, clusterName).Execute()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get global cluster: %w", err)
+	}
+	return geosharding.GetCustomZoneMapping(), nil
+}
+
+func (ds *ProductionAtlasDeployments) CreateCustomZones(ctx context.Context, projectID, clusterName string, mappings []akov2.CustomZoneMapping) (map[string]string, error) {
+	geosharding, _, err := ds.globalClusterAPI.CreateCustomZoneMapping(ctx, projectID, clusterName, customZonesToAtlas(&mappings)).Execute()
+	if err != nil {
+		return nil, fmt.Errorf("failed to create custom zone: %w", err)
+	}
+	return geosharding.GetCustomZoneMapping(), nil
+}
+
+func (ds *ProductionAtlasDeployments) DeleteCustomZones(ctx context.Context, projectID, clusterName string) error {
+	_, _, err := ds.globalClusterAPI.DeleteAllCustomZoneMappings(ctx, projectID, clusterName).Execute()
+	if err != nil {
+		return fmt.Errorf("failed to delete custom zone: %w", err)
+	}
+	return nil
+}
+
+func (ds *ProductionAtlasDeployments) GetZoneMapping(ctx context.Context, projectID, deploymentName string) (map[string]string, error) {
+	cluster, _, err := ds.clustersAPI.GetCluster(ctx, projectID, deploymentName).Execute()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get cluster: %w", err)
+	}
+	result := make(map[string]string, len(cluster.GetReplicationSpecs()))
+	for _, rc := range cluster.GetReplicationSpecs() {
+		result[rc.GetId()] = rc.GetZoneName()
+	}
+	return result, nil
+}
+
```

Comment:
If we'd prefer it converted I can do so, but since the Atlas API returns Custom Zones as a map I figured we could just stick with a native type. I appreciate this isn't strictly adhering to the idea behind the translation layer, but it seemed easier